builtin/fsmonitor--daemon.c

```diff
@@ -879,7 +879,7 @@ static int do_handle_client(struct fsmonitor_daemon_state *state,
 
 	shown = kh_init_str();
 	for (batch = batch_head;
-	     batch && batch->batch_seq_nr >= requested_oldest_seq_nr;
+	     batch && batch->batch_seq_nr + 1 >= requested_oldest_seq_nr;
```

Comment:
https://github.com/gitgitgadget/git/pull/923
https://github.com/gitgitgadget/git/pull/923/commits/be9e1396a630d5af8349c64c87ad4f277eb5b3af

Could you try the topmost TODO commit as an alternative and see if that also fixes the problem that you were seeing?

I noticed that your solution here always caused the oldest bin in the response to be repeated.

Reply:
I'll dress this up my commit later.

Reply:
The commit you gave fails the Scalar functional tests for me.